banderwagon/element.go

```diff
@@ -42,6 +42,41 @@ func (p Element) Bytes() [sizePointCompressed]byte {
 	return x.Bytes()
 }
 
+func BatchNormalize(elements []*Element) {
+	// The elements slice might contain duplicate pointers,
+	// dedupe them to avoid double work.
+	dedupedElements := make([]*Element, 0, len(elements))
+	for _, e := range elements {
+		found := false
+		for i := range dedupedElements {
+			if dedupedElements[i] == e {
+				found = true
+				break
+			}
+		}
+		if found {
+			continue
+		}
+		dedupedElements = append(dedupedElements, e)
+	}
```

Comment:
Deduping by doing lookups in a slice compared to a `map[*Element]struct{}`. I think we shouldn't have so many elements to justify a `map`.

Reply:
Seems the gnark-crypto method does not do this: https://github.com/ConsenSys/gnark-crypto/blob/441dc0ffe639294b8d09e394f24ba7575577229c/ecc/bls12-381/g1.go#L920

So we may need to keep this even if we switch to upstream gnark-crypto